examples/uart_esptool/README.md

```diff
@@ -0,0 +1,88 @@
+# RFC2217 UART Esptool Example
+
+This example sets up an RFC2217 server that emulates a serial port for esptool communication. It accepts client connections and forwards data between the RFC2217 client and UART, while also implementing the DTR/RTS control signals that esptool uses for programming ESP devices.
+
+The example is specifically designed to work with esptool for remote programming of ESP devices over the network. It implements the BOOT and EN GPIO control signals that esptool uses to put ESP devices into download mode.
+
+The example can be used with any ESP chip. You need to connect the target ESP device to the UART port of the ESP board running this example.
+
+Network connection is achieved using `protocols_examples_common` component, which provides a simple API for connecting to Wi-Fi or Ethernet. You can configure the connection method and the credentials in menuconfig.
+
+## How to Use the Example
+
+Build and flash the example as usual. For example, when using an ESP32-S3 chip:
+
+```shell
+idf.py set-target esp32s3
+idf.py flash monitor
+```
+
+By default, the example uses UART1, GPIO 4 for TX and GPIO 5 for RX. It also configures GPIO 26 as BOOT and GPIO 25 as EN for esptool control signals. If you want to use different UART or GPIOs, you can change the configuration in menuconfig.
+
+Note the IP address printed on the console when the example runs. You will need it to connect to the device. For example:
+
+```text
+I (4547) example_common: Connected to example_netif_sta
+I (4557) example_common: - IPv4 address: 192.168.0.196,
+```
+
+After flashing, the device will start an RFC2217 server. You can now use esptool to program a target ESP device remotely over the network.
+
+## Programming with Esptool
+
+Connect the target ESP device to the UART port of the ESP board running this example:
```

Comment:
Suggest splitting this part into its own section, "Hardware setup", placed before "How to Use the Example".